### PR TITLE
Added in SDCSSO for portal

### PR DESCRIPTION
**Pull request by qdzlug**
Hi @arekinath ,

Was doing this for customers in the demo boxes, so I decided it would be nice if I added in a bit to handle portal. Going to have a few other support-folks test this as well just to make sure I'm not doing anything dumb, but wanted to get a PR out there with what I have so far.

Jay

### CODE REVIEW

File: cns-hook [hunk 1]
```diff
@@ -73,6 +75,12 @@ function deploy_challenge {
 		verifyvm "$domain" "$vmuuid"
 		update_remotevm "$vmuuid" "$tokenval"
 		;;
+	sdcsso)
+		local admin=$(sdc-useradm get admin | json uuid)
+		vmuuid=$(sdc-vmadm list -H -o uuid owner_uuid="$admin" alias="sdcsso" | tail -1)
+		verifyvm "$domain" "$vmuuid"
+		update_remotevm "$vmuuid" "$tokenval"
```
Comment (arekinath) [inline cns-hook]: You've used `update_remotevm` here, but further down you're assuming that the `sdcsso` VM is on the headnode (by using `zlogin`, direct `cp`, etc.)

Do `sdcsso` zones normally get deployed directly on the headnode? If so, and we're ok with assuming it here, this should be an `update_localvm` call instead.

File: cns-hook [hunk 2]
```diff
@@ -125,6 +139,13 @@ function deploy_cert {
 		zlogin "${vmuuid}" svcadm restart adminui
 		echo "OK: adminui certificate deployed, and adminui restarted" >&2
 		;;
+	sdcsso)
+		vmuuid="$(vmadm lookup alias=sdcsso)"
+                cp "${certdir}/fullchain.pem" "/zones/${vmuuid}/root/opt/smartdc/sdcsso/ssl/cert.pem"
```
Comment (arekinath) [inline cns-hook]: The indentation here looks wrong

File: cns-hook [hunk 3]
```diff
@@ -191,6 +212,29 @@ function unchanged_cert {
...
+                        restart=true
+                fi
+
+                if ${restart} == true; then
+                        zlogin "${vmuuid}" svcadm restart sdcsso
+                        echo "OK: sdcsso certificate/key deployed, and sdcsso restarted" >&2
+                else
+                        echo "OK: sdcsso certificate/key up to date" >&2
+                fi
+		;;
```
Comment (arekinath) [inline cns-hook]: Indentation throughout this whole change here doesn't match what's already in the file (and doesn't match the ;; line either)

File: cns-hook [hunk 4]
```diff
@@ -125,6 +139,13 @@ function deploy_cert {
 		zlogin "${vmuuid}" svcadm restart adminui
 		echo "OK: adminui certificate deployed, and adminui restarted" >&2
 		;;
+	sdcsso)
+		vmuuid="$(vmadm lookup alias=sdcsso)"
+                cp "${certdir}/fullchain.pem" "/zones/${vmuuid}/root/opt/smartdc/sdcsso/ssl/cert.pem"
+                cp "${certdir}/privkey.pem" "/zones/${vmuuid}/root/opt/smartdc/sdcsso/ssl/key.pem"
+		zlogin "${vmuuid}" svcadm restart sdcsso
+		echo "OK: sdcsso certificate deployed, and sdcsso restarted" >&2
```
Comment (arekinath) [inline cns-hook]: It looks like these files are not kept between reprovisions of the sdcsso zone, is that correct? We might want to consider making a branch that injects these as server-side SAPI manifests then, as well, like we do for CloudAPI. That can be a separate thing though.

**Comment by qdzlug** [comment]
Hi @arekinath,

Thanks for looking into this....

So, I fixed the spacing and I think I managed to adjust the spacing across the board and not just hit my stuff. I just retab'd things to get rid of the tabs and replace with spaces. So - I may need a little guidance there on how to un-screw-up that bit. Most of my CR woes these days center around whitespace and spacing, so any advice there is greatly appreciated.

The other stuff - I changed the call to localvm from remotevm because we will be on the HN most of the time - the script we use lets you tell it where to provision things but we never advise anything but the HN (and we always use the HN in our demos). To be more clean I will look into a way to handle a remote SSO when I have some time and replace those commands (maybe it makes sense to look at doing this in conjunction with adding these to sapi)?

Jay

File: cns-hook (hunk 1 above)
Comment (bahamat) [reply cns-hook]: It's not always on the headnode. In two of our three deployments it's not on the headnode.

**Comment by bahamat** [comment]
@qdzlug Can you rebase this against the latest? In particular, all the functions form cns-hook have moved into cns-hook-util.

**Comment by bahamat** [comment]
Piranha itself now has all the capabilities of SSO, so this isn't needed anymore.

**Pull request closed without merging**